DevEdu.API/DevEdu.DAL/Repositories/LessonRepository.cs

```diff
@@ -103,51 +105,174 @@ public void AddTopicToLesson(int lessonId, int topicId)
+                           lessonEntry = lesson;
+                           lessonEntry.Teacher = teacher;
+                           lessonEntry.Topics = new List<TopicDto>();
+                           lessonEntry.Course = course;
+                           lessonDictionary.Add(lessonEntry.Id, lessonEntry);
+                       }
+
+                       lessonEntry.Topics.Add(topic);
+                       return lessonEntry;
+                   },
+                   new { teacherId},
+                   splitOn: "Id",
+                   commandType: CommandType.StoredProcedure
+               )
+               .Distinct()
+               .ToList();
+
+            return list;
         }
 
         public LessonDto SelectLessonById(int id)
         {
-            return _connection
-                .Query<LessonDto, UserDto, LessonDto>(
+            LessonDto result = new LessonDto();
+            _connection
+                .Query<LessonDto, UserDto, TopicDto, LessonDto>(
                     _lessonSelectByIdProcedure,
-                    (lesson, teacher) =>
+                    (lesson, teacher, topic)=>
                     {
-                        lesson.Teacher = teacher;
+                        result.Teacher = teacher;
+                        if(result.Topics == null)
+                        {
+                            result.Topics = new List<TopicDto>();
+                        }
+                        result.Topics.Add(topic);
                         return lesson;
                     },
                     new { id },
                     splitOn: "Id",
                     commandType: CommandType.StoredProcedure
                 )
                 .FirstOrDefault();
+
+            return result;
+        }
+
+        public LessonDto SelectLessonWithCommentsById(int id)
+        {
+            LessonDto result = new LessonDto();
+            _connection
+                .Query<LessonDto, UserDto, TopicDto, LessonDto>(
+                    _lessonSelectByIdProcedure,
+                    (lesson, teacher, topic) =>
+                    {
+                        result.Teacher = teacher;
+                        if (result.Topics == null)
+                        {
+                            result.Topics = new List<TopicDto>();
+                        }
+                        result.Topics.Add(topic);
+                        return lesson;
+                    },
+                    new { id },
+                    splitOn: "Id",
+                    commandType: CommandType.StoredProcedure
+                )
+                .FirstOrDefault();
+
+            result.Comments = SelectCommentsFromLessonByLessonId(id);
+
+            return result;
+        }
+
+        public LessonDto SelectLessonWithCommentsAndStudentsById(int id)
+        {
+            LessonDto result = SelectLessonWithCommentsById(id);
+
+            result.Students = SelectStudentsLessonByLessonId(id);
```

Comment:
you must do this on BLL since only one request to DB is possible in each repository method